### PR TITLE
feat(core): MIC-934 Moves logger out of internal folder

### PR DESCRIPTION
**Pull request by danielRicaud**

### CODE REVIEW

File: service/logger/logger.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 	"log/slog"
 	"os"
 
-	"github.com/opentdf/platform/service/internal/logger/audit"
+	"github.com/opentdf/platform/service/logger/audit"
```
Comment (dmihalcik-virtru) [inline service/logger/logger.go:10]: This doesn't just make the logger public - it also makes the audit package public, and exposes the internal structure of the `logger.Logger` type. And the audit package seems so tightly coupled with this that maybe they should be a single package with one object? Not sure, maybe we need the ability to extend or have mulitple audit objects?

Options I see:

1. Consolidate on just the audit name and move everything into that
2. replace the `Audit` field in logger with a (lowercase) `audit` and add an `Audit()` getter method or audit proxy methods to hide the internals of this type
3. Don't move everything out of internal. Instead create this as a new package and have `internal/logger/Logger` implement an interface defined here.

Who designed/is using the Audit stuff? I'll see if I can get them to review.

Comment (pbacon-blaber) [reply service/logger/logger.go:10]: I don't necessarily think that the internals of the Audit stuff is sensitive as I believe the audit events we're publishing are public?

That being said I think any of these would be fine. I had originally made all the audit stuff just part of the logger package but made it a "sub package" as it got more and more complex. I think it being separate also makes it easier to understand / add and edit existing audit events as well.

File: service/pkg/server/start.go [hunk 2]
```diff
@@ -11,18 +11,18 @@ import (
...
 
 const devModeMessage = `
 ██████╗ ███████╗██╗   ██╗███████╗██╗      ██████╗ ██████╗ ███╗   ███╗███████╗███╗   ██╗████████╗    ███╗   ███╗ ██████╗ ██████╗ ███████╗
 ██╔══██╗██╔════╝██║   ██║██╔════╝██║     ██╔═══██╗██╔══██╗████╗ ████║██╔════╝████╗  ██║╚══██╔══╝    ████╗ ████║██╔═══██╗██╔══██╗██╔════╝
-██║  ██║█████╗  ██║   ██║█████╗  ██║     ██║   ██║██████╔╝██╔████╔██║█████╗  ██╔██╗ ██║   ██║       ██╔████╔██║██║   ██║██║  ██║█████╗  
-██║  ██║██╔══╝  ╚██╗ ██╔╝██╔══╝  ██║     ██║   ██║██╔═══╝ ██║╚██╔╝██║██╔══╝  ██║╚██╗██║   ██║       ██║╚██╔╝██║██║   ██║██║  ██║██╔══╝  
+██║  ██║█████╗  ██║   ██║█████╗  ██║     ██║   ██║██████╔╝██╔████╔██║█████╗  ██╔██╗ ██║   ██║       ██╔████╔██║██║   ██║██║  ██║█████╗
+██║  ██║██╔══╝  ╚██╗ ██╔╝██╔══╝  ██║     ██║   ██║██╔═══╝ ██║╚██╔╝██║██╔══╝  ██║╚██╗██║   ██║       ██║╚██╔╝██║██║   ██║██║  ██║██╔══╝
 ██████╔╝███████╗ ╚████╔╝ ███████╗███████╗╚██████╔╝██║     ██║ ╚═╝ ██║███████╗██║ ╚████║   ██║       ██║ ╚═╝ ██║╚██████╔╝██████╔╝███████╗
-╚═════╝ ╚══════╝  ╚═══╝  ╚══════╝╚══════╝ ╚═════╝ ╚═╝     ╚═╝     ╚═╝╚══════╝╚═╝  ╚═══╝   ╚═╝       ╚═╝     ╚═╝ ╚═════╝ ╚═════╝ ╚══════╝
```
Comment (pbacon-blaber) [inline service/pkg/server/start.go:25]: also unusure if this whitespace was meant to be left inhere to format this block message or what

Comment (danielRicaud) [reply service/pkg/server/start.go:25]: Thanks @pbacon-blaber I'll restore the whitespace just to be safe!

File: service/logger/logger.go (hunk 1 above)
Comment (jrschumacher) [reply service/logger/logger.go:10]: There will be a need to extend audit in DSP if we want a uniform audit solution. How this is done is open to discussion.

Comment (strantalis) [reply service/logger/logger.go:10]: I think moving audit out is fine. What we need is a generic audit method that peps can call and pass the following struct.

https://github.com/opentdf/platform/blob/9c1053f154bb2b8b9d2170b677f84729f70ab64c/service/internal/logger/audit/utils.go#L18-L29

Comment (danielRicaud) [reply service/logger/logger.go:10]: Thank you all, for now should we leave audit exposed and follow back on adding that generic audit method in a follow-up task?

**Review by dmihalcik-virtru — APPROVED** [review]

**Review by jrschumacher — APPROVED** [review]

**Pull request merged**